README.md

```diff
@@ -47,6 +47,92 @@ Events can be tracked without explicit parameters:
 wt.track("pageview");
 ```
 
+# React Integration
+
+`@clutter/wt` includes a React integration that supports providing event parameters via React context, validating events, and hooks to easily track events and inspect the current context parameters.
+
+WT exports a default `WTProvider` component, and two hooks: `useTrack` and `useWT`. If
+additional customization is desired, there is also a `createProvider` utility which allows for custom typing and added
+event control. The components and hooks returned by each call to `createProvider` use a separate context and cannot be
+mixed with the default exports.
+
+```tsx
+type EventParams = WTEventParams & {
+  customParamA: string;
+  customParamB?: string;
+};
+
+// The returned provider and hooks share a single context and respect the EventParams type
+const { WTProvider, useTrack, useWT } = createProvider((params: EventParams) =>
+  wt.track("event", params)
+);
+```
+
+### Context
+
+Context providers may be nested to build event parameters contextually:
+
+```tsx
+<WTProvider params={{ pageName: "home" }}>
+  <WTProvider params={{ container: "hero" }}>
+    {/* Events tracked here will receive the params { pageName: 'home', container: 'hero' } */}
+  </WTProvider>
+  <WTProvider params={{ container: "body" }}>
+    {/* Events tracked here will receive the params { pageName: 'home', container: 'body' } */}
+  </WTProvider>
+</WTProvider>
+```
+
+### Hooks
+
+`useTrack` provides the most flexible interface to track an event in a context aware manner
+
+```tsx
+function List({ children }) {
+  // Parameters passed to `useTrack` will be merged into each call to `track`
+  const track = useTrack({ container: "list" });
```

Comment:
in the same vein as the above comment, it almost appears that you can use `useTrack` or `useWT` outside of a `WTProvider`. can we make the examples a bit more clear while still not having a ton of code to get the point across?

Reply:
I think code examples are best when they're very concise - I'm worried that adding extra context to each example will actually make them _less_ clear. How does amending the `React Integration` section to clarify how things must be composed sound?

Reply:
that sounds good!

Reply:
updated here: 9d1f29a576bb26d934d99e1299abbcef8a86eaad